modules/swagger-codegen/src/main/java/io/swagger/codegen/config/CodegenConfigurator.java

```diff
@@ -449,21 +461,31 @@ private static String toAbsolutePathStr(String path) {
 
     }
 
-    private void checkAndSetAdditionalProperty(String property, String propertyKey) {
+    private void checkAndSetAdditionalProperty(Object property, String propertyKey) {
         checkAndSetAdditionalProperty(property, property, propertyKey);
     }
 
-    private void checkAndSetAdditionalProperty(String property, String valueToSet, String propertyKey) {
-        if (isNotEmpty(property)) {
-            additionalProperties.put(propertyKey, valueToSet);
+    private void checkAndSetAdditionalProperty(Object property, Object valueToSet, String propertyKey) {
+        if (property instanceof String) {
+            if (isNotEmpty((String) property)) {
+                additionalProperties.put(propertyKey, valueToSet);
+            }
+        } else if (property instanceof List) {
+            if (null != property) {
+                additionalProperties.put(propertyKey, valueToSet);
+            }
         }
     }
 
     public static CodegenConfigurator fromFile(String configFile) {
 
         if (isNotEmpty(configFile)) {
             try {
-                return Json.mapper().readValue(new File(configFile), CodegenConfigurator.class);
+                if (configFile.endsWith(".json")) {
```

Comment:
This could be less redundant.

```java
Mapper mapper;
if (configFile.endsWith(".json")) {
  mapper Json.mapper();
} else if (configFile.endsWith(".yaml")) {
  mapper = Yaml.mapper();
}

return mapper.readValue(new File(configFile), CodegenConfigurator.class);
```